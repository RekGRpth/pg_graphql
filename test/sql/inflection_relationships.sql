begin;
    create view r as
        select
            distinct name
        from
            graphql.field
        where
            local_columns is not null
        order by
            name;

    savepoint o;

    create table account (
        id int primary key
    );

    create table blog_post(
        id int primary key,
        author_id  int,
        account_no  int,

        constraint fkey_author_id foreign key (author_id) references account(id),
        constraint fkey_author_no foreign key (account_no) references account(id)
    );

    savepoint a;

    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    -- test that first letter of capital type name is lower cased when used as a field
    alter table account rename to "AccOnunt";
    select * from r;

    savepoint a;

    -- Inflection off, Overrides: on
    comment on constraint fkey_author_id
        on blog_post
        is E'@graphql({"foreign_name": "ownerOO", "local_name": "Blogzzz"})';
    comment on constraint fkey_author_no
        on blog_post
        is E'@graphql({"foreign_name": "accountNO", "local_name": "NOblogz"})';
    select * from r;

    rollback to savepoint a;

    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select * from r;

    -- Inflection on, Overrides: on
    comment on constraint fkey_author_id
        on blog_post
        is E'@graphql({"foreign_name": "ownerOO", "local_name": "Blogzzz"})';
    comment on constraint fkey_author_no
        on blog_post
        is E'@graphql({"foreign_name": "accountNO", "local_name": "NOblogz"})';
    select * from r;

rollback;
