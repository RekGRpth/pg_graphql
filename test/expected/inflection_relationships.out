begin;
    create view r as
        select
            distinct name
        from
            graphql.field
        where
            local_columns is not null
        order by
            name;
    savepoint a;
    -- Inflection true, Overrides: off, Ends with '_id'
    comment on schema public is e'@graphql({"inflect_names": true})';
    create table account_holder (
        id int primary key,
        author_id int,
        constraint fkey_author_id foreign key (author_id) references account_holder(id)
    );
    select * from r;
          name           
-------------------------
 accountHolderCollection
 author
(2 rows)

    -- Inflection true, Overrides: off, does not end with '_id'
    rollback to savepoint a;
    comment on schema public is e'@graphql({"inflect_names": true})';
    create table account_holder (
        id int primary key,
        author int,
        constraint fkey_author_id foreign key (author) references account_holder(id)
    );
    select * from r;
          name           
-------------------------
 accountHolder
 accountHolderCollection
(2 rows)

    -- Inflection true, Overrides: true
    rollback to savepoint a;
    comment on schema public is e'@graphql({"inflect_names": true})';
    create table account_holder (
        id int primary key,
        account_id int,
        constraint fkey_account_id foreign key (account_id) references account_holder(id)
    );
    comment on constraint fkey_account_id
    on account_holder
    is E'@graphql({"foreign_name": "auTHor", "local_name": "children"})';
    select * from r;
   name   
----------
 auTHor
 children
(2 rows)

    -- Inflection false, Overrides: off, Ends with 'Id'
    rollback to savepoint a;
    comment on schema public is e'@graphql({"inflect_names": false})';
    create table "AccountHolder" (
        id int primary key,
        "authorId" int,
        constraint fkey_author_id foreign key ("authorId") references "AccountHolder"(id)
    );
    select * from r;
          name           
-------------------------
 accountHolderCollection
 author
(2 rows)

    -- Inflection false, Overrides: off, does not end with 'Id'
    rollback to savepoint a;
    comment on schema public is e'@graphql({"inflect_names": false})';
    create table "AccountHolder" (
        id int primary key,
        author int,
        constraint fkey_author_id foreign key (author) references "AccountHolder"(id)
    );
    select * from r;
          name           
-------------------------
 accountHolder
 accountHolderCollection
(2 rows)

    -- Inflection false, Overrides: true
    rollback to savepoint a;
    comment on schema public is e'@graphql({"inflect_names": false})';
    create table "AccountHolder"(
        id int primary key,
        "accountId" int,
        constraint fkey_account_id foreign key ("accountId") references "AccountHolder"(id)
    );
    comment on constraint fkey_account_id
    on "AccountHolder"
    is E'@graphql({"foreign_name": "auTHor", "local_name": "children"})';
    select * from r;
   name   
----------
 auTHor
 children
(2 rows)

rollback;
